src/stores/video.ts

```diff
@@ -342,76 +461,130 @@ export const useVideoStore = defineStore('video', () => {
   const updateLastProcessingUpdate = (recordingHash: string): void => {
     const info = unprocessedVideos.value[recordingHash]
     info.dateLastProcessignUpdate = new Date()
-    unprocessedVideos.value = { ...unprocessedVideos.value, ...{ [recordingHash]: info } }
+    unprocessedVideos.value = {
+      ...unprocessedVideos.value,
+      ...{ [recordingHash]: info },
+    }
   }
 
-  const processVideoChunksAndTelemetry = async (recordingHash: string, info: UnprocessedVideoInfo): Promise<void> => {
-    if (info.dateFinish === undefined) {
-      info.dateFinish = info.dateLastRecordingUpdate
-    }
+  // Progress tracking for video processing
+  const videoProcessingDetails = ref<VideoProcessingDetails>({})
+  const totalFilesToProcess = ref(0)
 
-    // eslint-disable-next-line jsdoc/require-jsdoc
-    const chunks: { blob: Blob; name: string }[] = []
+  const updateFileProgress = (filename: string, progress: number, message: string): void => {
+    videoProcessingDetails.value[filename] = { filename, progress, message }
+  }
+
+  const currentFileProgress = computed(() => {
+    return Object.values(videoProcessingDetails.value).map((detail) => ({
+      fileName: detail.filename,
+      progress: detail.progress,
+      message: detail.message,
+    }))
+  })
+
+  const overallProgress = computed(() => {
+    const entries = Object.values(videoProcessingDetails.value)
+    const totalProgress = entries.reduce((acc, curr) => acc + curr.progress, 0)
+    return (totalProgress / (totalFilesToProcess.value * 100)) * 100
+  })
+
+  const debouncedUpdateFileProgress = useDebounceFn((filename: string, progress: number, message: string) => {
+    updateFileProgress(filename, progress, message)
+  }, 100)
 
-    const dateStart = new Date(info.dateStart)
-    const dateFinish = new Date(info.dateFinish)
+  const processVideoChunksAndTelemetry = async (hashes: string[]): Promise<void> => {
+    totalFilesToProcess.value = hashes.length
+    videoProcessingDetails.value = {}
 
-    await tempVideoChunksDB.iterate((videoChunk, chunkName) => {
-      if (chunkName.includes(recordingHash)) {
-        chunks.push({ blob: videoChunk as Blob, name: chunkName })
+    const tasks = hashes.map(async (hash) => {
+      const info = unprocessedVideos.value[hash]
+      if (!info) return
+
+      const chunks: { blob: Blob; name: string }[] = []
+      if (info.dateFinish === undefined) {
+        info.dateFinish = info.dateLastRecordingUpdate
       }
-    })
+      debouncedUpdateFileProgress(info.fileName, 1, 'Processing video.')
 
-    // As we advance through the processing, we update the last processing update date, so consumers know this is ongoing
-    updateLastProcessingUpdate(recordingHash)
+      const dateStart = new Date(info.dateStart)
+      const dateFinish = new Date(info.dateFinish)
```

Comment:
This was not added. It was already here.

Reply:
As discussed in private, the differ on VSCode is being able to better point the changes, so we will use it to review this PR for now.